### DISCUSSION TITLE
[General] Sepolia transactions occasionally failing

### DISCUSSION THREAD

**Discussion by MichaelSheinmanCircle** [zkSync-Community-Hub/zksync-developers#548] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

Seeing an issue with sepolia transaction occasionally failing with an error "Error function_selector = 0x, data = 0x". This happens approximately once for every 50 transactions I send to the network. 

### Additional Details

Example transactions: 
https://sepolia.explorer.zksync.io/tx/0x7183effad349b128ef82d58eb27e6b5cdcd2d4f02ae37c8cf582a83136db1345
https://sepolia.explorer.zksync.io/tx/0xf9fb1d51b4492b288eade87f310a15c912439e3d7d52e3cc67a3953a98f2a5d6

I heard these issues tend to indicate gas problem, however I am using the `eth_gasPrice` and the `eth_estimateGas` RPC calls for the gas estimation, and also adding a buffer on these values for extra caution.

Expected behaviour would be either that the transaction fails with some error message when I send the transaction, or that the transaction succeeds. However, the current behaviour is that the sending part of the transaction goes through and then the transaction fails.

**Comment by bxpana** [comment]
Thank you for moving that 😁 Will have the team take a look and see if we can find something or hopefully someone else from the community having a similar issue can weigh in.

**Reply by bxpana** [reply]
@MichaelSheinman how are things going here? Have you run into these same issues on testnet lately?

**Reply by MichaelSheinmanCircle** [reply]
Yes, here is a recent example from 5 days ago: https://sepolia.explorer.zksync.io/tx/0x0d97de2763781cf36a7052d7655ef292327c59dccc84fe3747a89660b140b821

**Reply by MichaelSheinmanCircle** [reply]
Note that this case is a very odd race condition - our system will pick up the transaction from the blockchain successfully, and then for some reason the transaction will fail later before it's officially completed in ethereum. 
Perhaps a good start here is to update the error message shown on the explorer so we get some more descriptive information here.

**Comment by MichaelSheinmanCircle** [comment]
FYI got more failures than usual earlier today, not sure if it's related to an update  
https://sepolia.explorer.zksync.io/tx/0xbe205d1885163823ccf7ed939313808fbd01c9a05efeb93b2473f3fe893aee35
https://sepolia.explorer.zksync.io/tx/0xfae8be164bcf0b284338a5326bf0e3c9454083f2ddcd37454e9feafb78e1dce4
https://sepolia.explorer.zksync.io/tx/0xa8acc6eb7166b204f2ce514eae28326dfbcfae481fb78f5978fbd4addcb7d854

**Reply by bxpana** [reply]
Thank you, will add these to check. Right now we're not seeing anything out of the ordinary when replaying them on the `era-test-node`

**Reply by bxpana** [reply]
I'm seeing that the failed txns are using 100% of the gas no matter what the gas is set to, but shortly after the next txn is successful. Still investigating and will keep you guys posted.

**Reply by hbriese** [reply]
@bxpana any updates? Still seeing this quite frequently

**Reply by MichaelSheinmanCircle** [reply]
same

**Reply by bxpana** [reply]
@hbriese and @MichaelSheinman nothing yet. We're seeing some volatility in gas on Sepolia that we're trying to understand better. Right now the best suggestion is to increase the gas limit. We're looking into improving the `era-test-node` to help with understanding these txns.

**Reply by bxpana** [reply, marked as answer]
Circling back, this is usually caused by the underlying L1 gas spike, which causes the pubdata price to skyrocket. This then leads to the txn failing. Things seem to be more stable now, but if you start to see any issues again let us know and we can reopen this thread.
